src/Oci8/Oci8Connection.php

```diff
@@ -215,75 +217,137 @@ protected function getDoctrineDriver()
      * Usage: DB::executeFunction('function_name(:binding_1,:binding_n)', [':binding_1' => 'hi', ':binding_n' =>
      * 'bye'], PDO::PARAM_LOB).
      *
-     * @author Tylerian - jairo.eog@outlook.com
-     * @param string $sql (mixed)
-     * @param array $bindings (kvp array)
-     * @param int $returnType (PDO::PARAM_*)
-     * @param int $length
+     * @param string $functionName
+     * @param array  $bindings   (kvp array)
+     * @param int    $returnType (PDO::PARAM_*)
+     * @param int    $length
      * @return mixed $returnType
      */
-    public function executeFunction($sql, array $bindings = [], $returnType = PDO::PARAM_STR, $length = null)
+    public function executeFunction($functionName, array $bindings = [], $returnType = PDO::PARAM_STR, $length = null)
     {
-        $query = $this->getPdo()->prepare('begin :result := ' . $sql . '; end;');
+        $stmt = $this->createStatementFromFunction($functionName, $bindings);
 
         foreach ($bindings as $key => &$value) {
-            if (! preg_match('/^:(.*)$/i', $key)) {
-                $key = ':' . $key;
-            }
-
-            $query->bindParam($key, $value);
+            $stmt->bindParam($key, $value);
         }
 
-        $query->bindParam(':result', $result, $returnType, $length);
-        $query->execute();
+        $stmt->bindParam(':result', $result, $returnType, $length);
+        $stmt->execute();
 
         return $result;
     }
 
     /**
-     * Execute a PL/SQL Procedure and return its result.
+     * Execute a PL/SQL Procedure and return its results.
+     *
      * Usage: DB::executeProcedure($procedureName, $bindings).
      * $bindings looks like:
      *         $bindings = [
      *                  'p_userid'  => $id
      *         ];
      *
-     * @param string $procedureName
-     * @param array $bindings
-     * @param mixed $returnType
-     * @return array
+     * @param  string $procedureName
+     * @param  array  $bindings
+     * @return bool
      */
-    public function executeProcedure($procedureName, $bindings, $returnType = PDO::PARAM_STMT)
```

Comment:
i think the old method is not being used alot b/c the implementation was way to specific as mentioned in other issues.

I think we should add this to both versions...